shiny/workflowPlot/server.R

```diff
@@ -1,212 +1,57 @@
 options(shiny.maxRequestSize=100*1024^2) 
+
 # Define server logic
 server <- shinyServer(function(input, output, session) {
   bety <- betyConnect()
+
+  # Source Extra Funtions 
   source("workflowPlot_fcns.R", local = TRUE) # Load all functions that need to be defined for this script
   
+  # Sidebar
+  source("server_files/sidebar_server.R", local = TRUE)
   
-  # Update all workflow ids
-  observe({
-    # get_workflow_ids function (line 137) in db/R/query.dplyr.R takes a flag to check
-    # if we want to load all workflow ids.
-    # get_workflow_id function from query.dplyr.R
-    query <- isolate(shiny::parseQueryString(session$clientData$url_search))
-    all_ids <- get_workflow_ids(bety, query, all.ids=TRUE)
-    updateSelectizeInput(session, "all_workflow_id", choices = all_ids)
-  })
-  # Update all run ids
-  all_run_ids <- reactive({
-    # Retrieves all run ids for seleted workflow ids
-    # Returns ('workflow ',w_id,', run ',r_id)
-    req(input$all_workflow_id)
-    w_ids <- input$all_workflow_id
-    # Will return a list
-    run_id_list <- c()
-    for(w_id in w_ids){
-      # For all the workflow ids
-      r_ids <- get_run_ids(bety, w_id)
-      for(r_id in r_ids){
-        # Each workflow id can have more than one run ids
-        # ',' as a separator between workflow id and run id
-        list_item <- paste0('workflow ', w_id,', run ', r_id)
-        run_id_list <- c(run_id_list, list_item)
-      }
-    }
-    return(run_id_list)
-  })
-  # Update all run_ids ('workflow ',w_id,', run ',r_id)
-  observe({
-    updateSelectizeInput(session, "all_run_id", choices = all_run_ids())
-  })
-    
-  # Update variable names observeEvent on input$load 
-  observeEvent(input$load, {
-    req(input$all_run_id)
-    # All information about a model is contained in 'all_run_id' string
-    ids_DF <- parse_ids_from_input_runID(input$all_run_id)
-    var_name_list <- c()
-    for(row_num in 1:nrow(ids_DF)){
-      var_name_list <- c(var_name_list, var_names_all(bety, ids_DF$wID[row_num], ids_DF$runID[row_num]))
-    }
-    updateSelectizeInput(session, "variable_name", choices = var_name_list)
-  })
-  # Loads data for all workflow and run ids after the load button is pressed.
-  # All information about a model is contained in 'all_run_id' string
-  # Wrapper over 'load_data_single_run' in PEcAn.db::query.dplyr
-  # Model data different from observations data 
-  loadNewData <- eventReactive(input$load,{
-    req(input$all_run_id)
-    # Get IDs DF from 'all_run_id' string
-    ids_DF <- parse_ids_from_input_runID(input$all_run_id)
-    globalDF <- map2_df(ids_DF$wID, ids_DF$runID, ~load_data_single_run(bety, .x, .y))
-    return(globalDF)
-  })
+  # Page 1: Select Data
+  source("server_files/select_data_server.R", local = TRUE)
+  
+  # Page 2: Exploratory Plots
+  source("server_files/model_plots_server.R", local = TRUE)
+  source("server_files/model_data_plots_server.R", local = TRUE)
```

Comment:
Just out of curiosity -- is `local = TRUE` here required to make Shiny recognize these, per your question on Slack earlier? Based on the `source` documentation, it seems like you'd want this to be `FALSE` (the default), but since your demo clearly worked, I'm assuming this is correct. Might not be a bad idea to add somewhere into one of our Shiny documentation things that `local = TRUE` is required when `source`ing functions for Shiny apps.

Reply:
Where do you think I should add this in our documentation? It's in the [scoping section of the shiny documentation](https://shiny.rstudio.com/articles/scoping.html)

> If you want to split the `server` or `ui` code into multiple files, you can use `source(local = TRUE)` to load each file. You can think of this as putting the code in-line, so the code from the sourced files will receive the same scope as if you copied and pasted the text right there.